src/colorprint.h

```diff
@@ -17,12 +17,15 @@ enum LogColor {
   COLOR_WHITE
 };
 
-std::string FormatString(const char* msg, va_list args);
-std::string FormatString(const char* msg, ...);
+__attribute__((__format__(__printf__, 1, 0))) std::string FormatString(
+    const char* msg, va_list args);
+__attribute__((__format__(__printf__, 1, 2))) std::string FormatString(
+    const char* msg, ...);
 
-void ColorPrintf(std::ostream& out, LogColor color, const char* fmt,
-                 va_list args);
-void ColorPrintf(std::ostream& out, LogColor color, const char* fmt, ...);
+__attribute__((__format__(__printf__, 3, 0))) void ColorPrintf(
+    std::ostream& out, LogColor color, const char* fmt, va_list args);
+__attribute__((__format__(__printf__, 3, 4))) void ColorPrintf(
+    std::ostream& out, LogColor color, const char* fmt, ...);
```

Comment:
These `__attribute__` need to be hidden by preprocessor macro,
and only enabled for non-MSVC.

Reply:
Applied